src/algorithms/utils/TimeSeries.test.ts

```diff
@@ -1,6 +1,106 @@
 import { uniformDatesBetween, makeTimeSeries, updateTimeSeries } from './TimeSeries'
 import { interpolateTimeSeries } from '../run'
 
+import * as fs from 'fs'
+const { utils } = require('jest-snapshot')
+
+expect.extend({
+  toBeCloseToArraySnapshot(received: number[]) {
+    const { testPath, currentTestName, snapshotState } = this
+    snapshotState._counters.set(currentTestName, (snapshotState._counters.get(currentTestName) || 0) + 1)
+    const count = Number(snapshotState._counters.get(currentTestName))
+    const key = utils.testNameToKey(currentTestName, count)
+    const expected = snapshotState._snapshotData[key]
+
+    /* If this isn't done, Jest reports the test as 'obsolete' and prompts for deletion. */
+    snapshotState._uncheckedKeys.delete(key)
+
+    /* If the snapshot isn't JSON, then return an empty array. Otherwise
+     * the test will fails in an inscrutable way.
+     */
+    const tryParse = (str: string) => {
+      try {
+        return JSON.parse(str)
+      } catch (e) {
+        return []
+      }
+    }
+
+    /* If the expected field is blank, then this is probably a new snapshot. */
+    const expectedDeserialized = expected ? tryParse(expected) : []
```

Comment:
Try to parse the expected string into, we assume, an array of floating point numbers. This commit is a hack, so no sanity checks.